File: Swap Nodes in Pairs LC-24.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def swapPairs(self, head: ListNode) -> ListNode:
        if not head:
          return
        
        dummy=ListNode(0,head)
        prev=dummy
        curr=next=dummy.next
        
        while next and next.next:
          next=curr.next.next
          prev.next=curr.next
          prev.next.next=curr
          curr.next=next
          prev=curr
          curr=next
          
        return dummy.next

File: Swap Nodes in Pairs LC-24.cpp
// tc=o(n)
// sc=o(1)
class Solution {
public:
    ListNode* swapPairs(ListNode* head) {
        if(!head || !head->next) return head;
        ListNode* dummy=new ListNode();
        dummy->next=head;
        head=dummy;
        while(head->next && head->next->next){
            ListNode* temp=head->next;
            head->next=temp->next;
            head=head->next;
            temp->next=head->next;
            head->next=temp;
            head=head->next;
        }
        return dummy->next;
    }
};